spec/models/time_period_spec.rb

```diff
@@ -0,0 +1,43 @@
+# == Schema Information
+#
+# Table name: time_periods
+#
+#  id         :bigint           not null, primary key
+#  end_date   :date
+#  start_date :date
+#  created_at :datetime         not null
+#  updated_at :datetime         not null
+#
+require 'rails_helper'
+
+RSpec.describe TimePeriod, type: :model do
+  let!(:time_period3) { FactoryBot.create(:time_period, start_date: Date.current - 8.weeks, end_date: Date.current - 7.weeks) }
+  let!(:time_period2) { FactoryBot.create(:time_period, start_date: Date.current - 6.weeks, end_date: Date.current - 5.weeks) }
+  let!(:time_period1) { FactoryBot.create(:time_period, start_date: Date.current - 4.weeks, end_date: Date.current - 3.weeks) }
+  let!(:time_period) { create :time_period }
```

Comment:
Please configure the factory using sequencing so that you can build multiple ones in the same test like so:
```ruby
let!(:time_period1) { create(:time_period) }
let!(:time_period2) { create(:time_period) }
```